File: archunit/solution/ArchUnit-kata/Layered/Controllers/SuperHeroController.cs
using ArchUnit.Kata.Layered.Repositories;
using ArchUnit.Kata.Layered.Services;

namespace ArchUnit.Kata.Layered.Controllers
{
    public class SuperHeroController
    {
        private readonly SuperHeroService superHeroService;
        private readonly ISuperHeroRepository superHeroRepository;

        public SuperHeroController(SuperHeroService superHeroService,
            ISuperHeroRepository superHeroRepository)
        {
            this.superHeroService = superHeroService;
            this.superHeroRepository = superHeroRepository;
        }
    }
}

File: archunit/solution-java/src/main/java/archunit/layered/api/SuperHeroController.java
package archunit.layered.api;

import archunit.layered.domain.SuperHeroRepository;
import archunit.layered.service.SuperHeroService;

public class SuperHeroController {
    private final SuperHeroService superHeroService;
    private final SuperHeroRepository superHeroRepository;

    public SuperHeroController(SuperHeroService superHeroService, SuperHeroRepository superHeroRepository) {
        this.superHeroService = superHeroService;
        this.superHeroRepository = superHeroRepository;
    }
}
